src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -4532,6 +4532,10 @@ internal override string FormatTypeName(bool serialization)
             }
         }
 
+        // This method looks like an attractive inline but expands to two calls,
+        // neither of which can be inlined or optimized further. So block it
+        // from inlining.
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
It would be nice if in the future the JIT could discover this fact and have a way to persist it so the next time we try to ngen/jit it we consult the database of known methods for this kind of information.  
Perhaphs could add an option to ngen/crossgen to create a simplist .IBC data file that contained this information and then check it in along with the real .IBC data.

Reply:
I like the idea but do not know how we'd actually do something like this.

I think it might be easier to improve the inlining heuristics in this area (a relatively small method body has multiple sequential calls that are not themselves inline candidates).  A strong enough conviction here would automatically propagate noinline into the runtime metadata when prejitting.

Reply:
Opened #20526 to track this idea.